File: Olympiad/COCI/official/2008/contest3_solutions/dejavu.cpp
#include <cstdio>

int n;
int x[100000], y[100000];
int freqX[100001], freqY[100001];

int main( void ) {
   scanf( "%d", &n );

   for( int i = 0; i < n; ++i ) {
      scanf( "%d%d", &x[i], &y[i] );
      ++freqX[ x[i] ];
      ++freqY[ y[i] ];
   }

   long long ret = 0;
   for( int i = 0; i < n; ++i )
      ret += (freqX[ x[i] ] - 1LL) * (freqY[ y[i] ] - 1LL);

   printf( "%lld\n", ret );

   return 0;
}

File: Olympiad/COCI/official/2008/contest3_solutions/dejavu.pas
program dejavu;

var n, i : longint;
    x, y : array [1..100000] of longint;
    freqX, freqY : array[1..100000] of longint;
    ret : int64;

begin
   readln( n );

   for i := 1 to 100000 do begin
      freqX[i] := 0;
      freqY[i] := 0;
   end;

   for i := 1 to n do begin
      readln( x[i], y[i] );
      freqX[ x[i] ] := freqX[ x[i] ] + 1;
      freqY[ y[i] ] := freqY[ y[i] ] + 1;
   end;

   ret := 0;
   for i := 1 to n do
      ret := ret + int64(freqX[ x[i] ]-1) * int64(freqY[ y[i] ]-1);

   writeln( ret );
end.
